modules/apps/translation/translation-service/src/main/java/com/liferay/translation/internal/search/spi/model/index/contributor/TranslationEntryModelDocumentContributor.java

```diff
@@ -38,6 +49,45 @@ public void contribute(
 		document.addText(Field.CONTENT, translationEntry.getContent());
 		document.addDate(
 			Field.MODIFIED_DATE, translationEntry.getModifiedDate());
+		document.addKeyword(Field.STATUS, translationEntry.getStatus());
+
+		String title = _getTitle(translationEntry);
+
+		document.addText(Field.TITLE, title);
 	}
 
+	private Locale _getLocale(TranslationEntry translationEntry)
+		throws PortalException {
+
+		ServiceContext serviceContext =
+			ServiceContextThreadLocal.getServiceContext();
+
+		if (serviceContext == null) {
+			return _portal.getSiteDefaultLocale(translationEntry.getGroupId());
+		}
+
+		return serviceContext.getLocale();
+	}
+
+	private String _getTitle(TranslationEntry translationEntry) {
+		try {
+			AssetRendererFactory<?> assetRendererFactory =
```

Comment:
Hi @adolfopa,

Shouldn't we be doing this with the Info API?

Reply:
The problem is that (afaik), the Info API doesn't provide a quick and efficient way of getting the title; we need to create the whole set of infoValues only to get the title. Here we're always looking for the TranslationEntry asset renderer, so there's no danger (it is guaranteed that an implementation will exist).

Reply:
Also, the infoValues for the TranslationEntry currently *don't* include the title, and I don't think we want to do that (it could mess with translation importing).

Reply:
Makes sense. I will write it down and talk to echo about this to see if it makes sense to the info framework at some point.